DataManagementSystem/Client/FailoverTransfer.py

```diff
@@ -72,18 +75,33 @@ def transferAndRegisterFile(self,
     fileChecksum = fileMetaDict.get("Checksum", None)
 
     for se in destinationSEList:
-      self.log.info("Attempting dm.putAndRegister",
-                    "('%s','%s','%s',guid='%s',catalog='%s', checksum = '%s')" %
-                    (lfn, localPath, se, fileGUID, fileCatalog, fileChecksum))
-
-      result = DataManager(
-          catalogs=fileCatalog,
-          masterCatalogOnly=masterCatalogOnly).putAndRegister(lfn,
-                                                              localPath,
-                                                              se,
-                                                              guid=fileGUID,
-                                                              checksum=fileChecksum)
-      self.log.verbose(result)
+
+      # We put here some retry in case the problem comes from the FileCatalog
+      # being unavailable. If it is, then the `hasAccess` call would fail,
+      # and we would not make any failover request. So the only way is to wait a bit
+      # This keeps the WN busy for a while, but at least we do not lose all the processing
+      # time we just spent
+      for sleeptime in (10, 60, 300, 600):
+        self.log.info("Attempting dm.putAndRegister",
+                      "('%s','%s','%s',guid='%s',catalog='%s', checksum = '%s')" %
+                      (lfn, localPath, se, fileGUID, fileCatalog, fileChecksum))
+
+        result = DataManager(
+            catalogs=fileCatalog,
+            masterCatalogOnly=masterCatalogOnly).putAndRegister(lfn,
+                                                                localPath,
+                                                                se,
+                                                                guid=fileGUID,
+                                                                checksum=fileChecksum)
+        self.log.verbose(result)
+
+        # If the FC is unavailable, we stay in the loop and retry
+        # otherwise we continue without retrying
+        if result['OK'] or not cmpError(result, EFCERR):
+          break
+        self.log.error("transferAndRegisterFile: FC unavailable, retry")
+        time.sleep(sleeptime)
```

Comment:
Is it necessary to disable the JobWrapper Watchdog CPU consumption monitoring here?

Reply:
I must admit being totally unknowledgeable here, but I was under the impression that the download/upload were always done with the Watchdog disabled ?

Reply:
Indeed.
https://gitlab.cern.ch/lhcb-dirac/LHCbDIRAC/-/blob/master/LHCbDIRAC/Workflow/Modules/UploadOutputData.py#L232-233

https://gitlab.cern.ch/CLICdp/iLCDirac/ILCDIRAC/-/blob/Rel-v29r2/Workflow/Modules/UploadOutputData.py#L180-183

But not in DIRAC itself, as far as I can tell.

Reply:
ah ok. Well if it isn't disabled before, I don't see why i should be now. Also, this should be an extremely rare case !

Reply:
Yes, agreed. The disabling needs to happen at a higher level than this function here.